merkletree/nodekv_test.go

```diff
@@ -0,0 +1,113 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/coniks-go/storage/kv"
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+func TestNodeSerializationAndDeserialization(t *testing.T) {
+	util.WithDB(func(db kv.DB) {
+		m, err := NewMerkleTree()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		key1 := "key1"
+		val1 := []byte("value1")
+		key2 := "key3"
+		val2 := []byte("value2")
+		index1 := vrfPrivKey1.Compute([]byte(key1))
+		if err := m.Set(index1, key1, val1); err != nil {
+			t.Fatal(err)
+		}
+
+		index2 := vrfPrivKey1.Compute([]byte(key2))
+		if err := m.Set(index2, key2, val2); err != nil {
+			t.Fatal(err)
+		}
+
+		m.recomputeHash()
+
+		ap := m.Get(index1)
+		if ap.Leaf.Value() == nil {
+			t.Fatal("Cannot find key:", key1)
+		}
+
+		enWant, ok := m.root.leftChild.(*emptyNode)
+		if !ok {
+			t.Fatal("Bad type insertion")
+		}
+		inWant, ok := m.root.rightChild.(*interiorNode)
+		if !ok {
+			t.Fatal("Bad type insertion")
+		}
+		lnWant := ap.Leaf.(*userLeafNode)
+
+		wb := db.NewBatch()
+		enWant.storeToKV(1, []bool{false}, wb)
+		inWant.storeToKV(1, []bool{true}, wb)
+		lnWant.storeToKV(1, util.ToBits(lnWant.index)[:lnWant.level], wb)
```

Comment:
Is ``util.ToBits(lnWant.index)[:lnWant.level]`` how the developer is going to have to call ``ln.storeToKV()``? If so, I think the interface should abstract this detail away.